homeassistant/components/unmanic/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""Sensor platform for Unmanic."""
+from homeassistant.components.sensor import SensorEntity
+
+from .const import DOMAIN, NAME
+from .entity import UnmanicEntity
+
+SENSORS = ["total-workers", "active-workers", "queue", "history"]
+
+
+async def async_setup_entry(hass, entry, async_add_devices):
+    """Unmanic setup sensor entity."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_devices([UnmanicSensor(coordinator, entry, key) for key in SENSORS])
+
+
+class UnmanicSensor(UnmanicEntity, SensorEntity):
+    """Unmanic Sensor class."""
+
+    def __init__(self, coordinator, config_entry, key) -> None:
+        """Initialize."""
+        self.coordinator = coordinator
+        self.config_entry = config_entry
+        self.key = key
+
+        super().__init__(
+            coordinator=coordinator,
+            config_entry=config_entry,
+            device_id=config_entry.entry_id,
+        )
+
+    @property
+    def unique_id(self):
+        """Return a unique ID to use for this entity."""
+        return f"{self.config_entry.entry_id}_{self.key}"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self.key == "total-workers":
```

Comment:
I am not a huge fan of all the `if/else` statements in this entity. Are you familiar with the Entity Descriptions? See tuya for an example: https://github.com/home-assistant/core/blob/dev/homeassistant/components/tuya/sensor.py